time-based-packet-transmission-offload.md

```diff
@@ -153,6 +171,29 @@ typedef enum _NDIS_TXTIME_SUPPORT_FLAGS {
 
 ```
 
+To obtain the `NDIS_TCP_IP_TXTIME_NET_BUFFER_LIST_INFO` structure, a driver should call the `NET_BUFFER_LIST_INFO` macro with an _Id of `TcpIpTxTimeNetBufferListInfo`
+```C
+typedef struct _NDIS_TCP_IP_TXTIME_NET_BUFFER_LIST_INFO {
+  struct {
+    ULONG IsIPv4 : 1;
+    ULONG IsIPv6 : 1;
+    ULONG IsTcp : 1;
+    ULONG IsUdp : 1;
```

Comment:
Why would the NIC want to know it's a V4 or V6 packet? The same goes for TCP and UDP too.

Reply:
If we can, I definitely would prefer to eliminate this extra stuff too.

Reply:
Or can we leverage existing OOBs? E.g. reuse some checksum offload fields? It's crazy that each offload has an "IsIpv6" flag...